spec/models/user_spec.rb

```diff
@@ -127,6 +127,58 @@
     end
   end
 
+  describe "#update" do
+    let(:existing_user) { create(:user) }
+
+    it "sets renamed_at if username changed" do
+      existing_user.update(login: "new_username")
+      expect(existing_user.renamed_at).to be_within(1.minute).of(Time.now.utc)
```

Comment:
You can [freeze time](https://api.rubyonrails.org/v5.2/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-freeze_time) here as well. Then you can just check that it's equal to `Time.now`.